Framework/Core/Windows/Devices/Gpio/GpioController.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Windows.Devices.Gpio
+{
+    public sealed class GpioController
+    {
+        // Public properties
+
+        extern public int PinCount
+        {
+            [MethodImplAttribute(MethodImplOptions.InternalCall)]
+            get;
+        }
+
+        // Public methods
+
+        public static GpioController GetDefault()
+        {
+            return s_instance;
+        }
+
+        public GpioPin OpenPin(int pinNumber)
+        {
+            return OpenPin(pinNumber, GpioSharingMode.Exclusive);
+        }
+
+        public GpioPin OpenPin(int pinNumber, GpioSharingMode sharingMode)
+        {
+            return new GpioPin(pinNumber);
+        }
+
+        public bool TryOpenPin(int pinNumber, GpioSharingMode sharingMode, out GpioPin pin, out GpioOpenStatus openStatus)
+        {
+            pin = null;
+
+            try
+            {
+                pin = OpenPin(pinNumber, sharingMode);
+            }
+            catch

```

Comment:
Don't catch all exceptions here. Create an overlod (probably private) to OpenPin that has a flag on whether it should throw exceptions or not. Tryxxx functions don't guarantee no exceptions can occur, they only guarantee that any exceptions known to the implementation won't be thrown. There are still critical and runtime asynchronous exceptions that can occur at any point (i.e. ThreadAbortException) etc... Catching all exceptions like this looses those and alters expected behavior of the system.